autoload/handlers.vim

```diff
@@ -331,19 +331,26 @@ def s:processHoverReply(lspserver: dict<any>, req: dict<any>, reply: dict<any>):
     return
   endif
   if lspOptions.hoverInPreview
-      silent! pedit HoverReply
-      wincmd P
-      setlocal buftype=nofile
-      setlocal bufhidden=delete
-      if !reply.result.contents.kind->empty()
-          exe 'setlocal ft=' .. reply.result.contents.kind
-      endif
-      deletebufline(bufnr(), 1,  getbufinfo(bufnr())[0].linecount)
-      append(0, hoverText)
-      cursor(1, 1)
-      wincmd p
+    silent! pedit HoverReply
+    wincmd P
+    setlocal buftype=nofile
+    setlocal bufhidden=delete
+    # If lsp server provide kind of hover reply and this kind is in syntaxs
+    # syported by vim, use it for file type of preview window. Else use
+    # type text by default.
+    if reply.result.contents->has_key('kind')
+\        && !reply.result.contents.kind->empty()
+\        && getcompletion('', 'filetype')->index(reply.result.content.kind) > -1
+      exe 'setlocal ft=' .. reply.result.contents.kind
```

Comment:
what i meant in that comment about ft syntax was e.g hovertext for bash perhaps was  `markdown` with `man` type content, maybe had syntax for such `man` content (though/and in `markdown` kind/ft), but that maybe not highlighted initially before switched into preview window, though not very sure why or maybe not much people care about that...

Reply:
This is a suggestion. May be we can add an option for an lsp server to force the type of syntax to use with hover reply for special case with something like :

Example of server config : 
```vim
call lsp#addServer({
\       'filetype': 'bash',
\       'path': path/to/lsp',
\       'hoverReplyKind': 'man' " <- force to set syntax type for hover reply to 'man' for the bash lsp
\ })
```

In `autoload/handler.vim` for `s:processHoverReply` : 
```vim
# If we set the configuration of the server to use specific file type to display reply, use this type.
if lspserver->has_key('hoverReplyKind') && ! lspserver.hoverReplyKind->empty() && getcompletion('', 'filetype')->index(lspserver.hoverReplyKind) > -1
  hoverKind = lspserver.hoverReplyKind
else
  ...
endif
```

Reply:
i think IF it is a 'markdown' kind:
1, if it is a '```foo' such kind markdown content, then to do as following, or just a 'markdown';
2, directly get ft from its markdown mark e.g '```man' then means 'man', or just a 'markdown';
3, get rid (or conceal) of first line for this mark (and the last line '```')
it can write a separate func for this, we can the same thing for the popup hover as well if so.
// just like what i tried in pr #4, but just now we can set the ft both for lsphover preview and lsphover popup per this func.

Reply:
or whatever, above you did for preview lsphover perhaps was good enough for now.
// i've done such logic for popup lsphover at my local, and actually not much lsp server offerred markdown kind hover doc.